bin/bats

```diff
@@ -3,42 +3,70 @@
 set -e
 
 BATS_READLINK='true'
+BATS_REALPATH=''
 if command -v 'greadlink' >/dev/null; then
   BATS_READLINK='greadlink'
 elif command -v 'readlink' >/dev/null; then
   BATS_READLINK='readlink'
 fi
 
+set +e
+$BATS_READLINK -m $0 &>/dev/null;
+[ $? -ne 0 ] && BATS_REALPATH='realpath'
+set -e
+
+resolve_path() {
+  local path=$1
+  local target_path;
+
+  if [[ -L $path ]]; then
+    # Busybox has no readlink from coreutisl
+    if [[ ! -z "$BATS_REALPATH" ]]; then
+      target_path="$(realpath "$path")"
+      # we need to traverse the paths
+      target_path="$(dirname "$target_path")" # strip bats
+      target_path="$(dirname "$target_path")" # strip bin
+    else
+      target_path="$("$BATS_READLINK" -m "$path/../../")"
+    fi
+  elif [[ -f $path ]]; then
+    target_path=$(dirname $path);
+    if [[ ! -z "$BATS_REALPATH" ]]; then
+      target_path="$(realpath "$target_path")"
```

Comment:
As above, should this use the variable rather than invoking `realpath` directly?